server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import glob
+import re
+import errno
+import hashlib
+import shutil
+import tempfile
+import tarfile
+from enum import Enum
+from argparse import ArgumentParser
+from s3backup import S3Config, Entry
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger, md5sum
+from pbench.report import Report
+
+
+_NAME_ = "pbench-audit-server"
+
+def verify_subdirs(config, controller, cnt):
+
+    linkdirs = sorted(config.LINKDIRS.split(' '))
+
+    states = glob.iglob(os.path.join(controller,'*'))
+    tarball, symlink, unexpected_objects, unexpected_dirs = ([] for i in range(4))
+
+    # Checking if the list is empty or not
+    if len(os.listdir(controller)) == 0:
```

Comment:
Is there a reason why we can't just check if `states` is empty?

Reply:
It is a genrerator, so it will need to be forced in order to calculate the length. But maybe the code can be reorganized toa allow us to use the generator without forcing it.

Reply:
Generator can be used only once